### PR TITLE
[Communication] - PhoneNumberAdministrationClient - fix next page

### PR DESCRIPTION
**Pull request by pavelprystinka**
Fixing issue with Pageable and AsyncPageable results are returning only a first page of result set.

### CODE REVIEW

File: sdk/communication/Azure.Communication.Administration/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 ## 1.0.0-beta.4 (Unreleased)
 
+### Fixed
+- Issue with paging results not pulling next pages
```
Comment (DominikMe) [inline sdk/communication/Azure.Communication.Administration/CHANGELOG.md:6]: Thanks for adding this 👍

File: sdk/communication/Azure.Communication.Administration/tests/PhoneNumberAdministrationClient/PhoneNumberAdministrationClientLiveTests.cs [hunk 2]
```diff
@@ -58,6 +58,17 @@ public async Task GetAllPhoneNumbers()
...
 
+        [Test]
+        public async Task GetAllReservations()
+        {
+            var client = CreateClient();
+
+            var reservationsPagable = client.GetAllReservationsAsync();
+            var reservations = await reservationsPagable.ToEnumerableAsync();
+
+            Assert.IsNotEmpty(reservations);
```
Comment (DominikMe) [inline sdk/communication/Azure.Communication.Administration/tests/PhoneNumberAdministrationClient/PhoneNumberAdministrationClientLiveTests.cs:69]: Can we guarantee that this is not empty on a fresh created resource and where we can't guarantee the order in which the tests execute?

Consider `IsNotNull` if we're not sure.

**Review by DominikMe — APPROVED** [review]
Lgtm, thanks for fixing!

File: sdk/communication/Azure.Communication.Administration/src/PhoneNumberAdministrationClient.cs [hunk 3]
```diff
@@ -65,10 +65,13 @@ public virtual AsyncPageable<AcquiredPhoneNumber> GetAllPhoneNumbersAsync(string
 
             try
             {
-                return PageResponseEnumerator.CreateAsyncEnumerable(async s =>
+                return PageResponseEnumerator.CreateAsyncEnumerable(async nextLink =>
                 {
-                    Response<AcquiredPhoneNumbers> response = await RestClient.GetAllPhoneNumbersAsync(locale, skip: null, take: null, cancellationToken).ConfigureAwait(false);
-                    return Page.FromValues(response.Value.PhoneNumbers, continuationToken: null!, response.GetRawResponse());
+                    Response<AcquiredPhoneNumbers> response = string.IsNullOrEmpty(nextLink)
```
Comment (RezaJooyandeh) [inline sdk/communication/Azure.Communication.Administration/src/PhoneNumberAdministrationClient.cs:70]: we can just do the null check instead `nextLink is null`

**Review by RezaJooyandeh — APPROVED** [review]
Thank you for fixing this. Looks good to me with on only 1 comment that shows up in a few places. Feel free to merge after updating it

File: sdk/communication/Azure.Communication.Administration/tests/PhoneNumberAdministrationClient/PhoneNumberAdministrationClientLiveTests.cs (hunk 2 above)
Comment (pavelprystinka) [reply sdk/communication/Azure.Communication.Administration/tests/PhoneNumberAdministrationClient/PhoneNumberAdministrationClientLiveTests.cs:88]: I have added CreateReservation step to the test so there will be at least one reservation

**Pull request merged**